From 8efc7d127ab2189a44bc64b862c9feebf82b73f3 Mon Sep 17 00:00:00 2001
From: Marius Brehler <marius.brehler@amd.com>
Date: Fri, 21 Mar 2025 17:01:20 +0000
Subject: [PATCH 2/7] Work around race condition
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

With `add_dependency`, compiling the `hipblas_fortran_client` target
fails with "Cannot open module file ‘hipblas.mod’ for reading at (1): No
such file or directory" as `hipblas.mod` is created in parallel but not
in time for the first run. Using `target_link_libraries` resovles this.
---
 clients/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/clients/CMakeLists.txt b/clients/CMakeLists.txt
index 9154cce..86b470b 100644
--- a/clients/CMakeLists.txt
+++ b/clients/CMakeLists.txt
@@ -99,7 +99,7 @@ if( BUILD_CLIENTS_TESTS OR BUILD_CLIENTS_BENCHMARKS OR BUILD_CLIENTS_SAMPLES )
     else()
       add_library(hipblas_fortran_client STATIC ${hipblas_f90_source_clients_no_solver})
     endif()
-    add_dependencies(hipblas_fortran_client hipblas_fortran)
+    target_link_libraries(hipblas_fortran_client hipblas_fortran)
   endif()
   include_directories(${CMAKE_BINARY_DIR}/include/hipblas)
   include_directories(${CMAKE_BINARY_DIR}/include)
-- 
2.47.1.windows.2

